### PR TITLE
Modified DataServlet to return hard-coded JSON data and script to fetch that JSON to build UI

### PR DESCRIPTION
**Pull request by thuavu**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -19,14 +19,26 @@
...
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Thu!</h1>");
+    //response.setContentType("text/html;");
+    //response.getWriter().println("<h1>Hello Thu!</h1>");
```
Comment (TsueiCat) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Remove commented-out lines. Same goes for other files.

File: portfolio/pom.xml [hunk 2]
```diff
@@ -23,6 +23,12 @@
       <version>4.0.1</version>
       <scope>provided</scope>
     </dependency>
+
+    <dependency>
+        <groupId>com.google.code.gson</groupId>
+        <artifactId>gson</artifactId>
+        <version>2.8.6</version>
+    </dependency>
```
Comment (TsueiCat) [inline portfolio/pom.xml]: Can you get rid of the TODO comment on line 43?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -19,14 +19,26 @@
...
+    response.setContentType("application/json;");
+
+    ArrayList<String> listValues = new ArrayList<String>();
```
Comment (TsueiCat) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: listValues doesn't seem to accurately describe what the ArrayList. comments would be a better name. Also, when initializing the ArrayList, 'new ArrayList<>()' will do the trick

**Review by TsueiCat — CHANGES_REQUESTED** [review]
Could you also add a screenshot of the changes to the description?

File: portfolio/pom.xml (hunk 2 above)
Comment (thuavu) [reply portfolio/pom.xml]: Yep

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (thuavu) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Will do.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (thuavu) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: image
@TsueiCat Got it

**Comment by thuavu** [comment]
image
image
@TsueiCat Can do.

**Review by TsueiCat — CHANGES_REQUESTED** [review]
Hi Thu! Could you add those commits with the changes made to this pull request?

**Comment by thuavu** [comment]
@TsueiCat Uh, at the moment I already went ahead with doing the step 4 stuff so it would commit the code I have now

**Review by TsueiCat — COMMENTED** [review]
I see. Hopefully, you committed those changes as separate commits and would be able to cherry pick them to this pull request. See this article for more details:
I think you can move these commits to the first pull request branch using cherry-pick. Here's an article explaining more in detail.
https://dev.to/projectpage/how-to-move-a-commit-to-another-branch-in-git-4lj4

Alternatively, since the changes to make are not too complicated you could just copy paste them from the branch you have them to this branch and push the commits to this PR and the other one as needed. I know this is tedious, but it's good practice using git and getting more comfortable with how the PR review process works :)

**Comment by thuavu** [comment]
The last two commits are in the wrong branch (64cdada and c20f21d)

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -26,3 +26,15 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+/* Request content from the server and add it to the html page */
+function requestContent() {
+    /*
```
Comment (TsueiCat) [inline portfolio/src/main/webapp/script.js]: Get rid of this comment-out code :)

**Review by TsueiCat — APPROVED** [review]
One more deletion and you can merge!

**Pull request merged**